Remove some config properties with adding appropriate values to the codebase

lkbhitesh07 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of N/A.
2. This PR does the following: This PR replaces some of the config properties with the appropriate value in the codebase. The config properties are:
(a) VMID_SHARED_SECRET_KEY_MAPPING: I will request the admin to put the value in cloud secrets and we will be accessing it from there.
(b) FEATURED_TRANSLATION_LANGUAGES: Shift the value to the constants.ts file. This value is currently empty but we do have a handler for this value to use in the frontend and that's why I plan to shift it to constants.ts
(c) LIST_OF_DEFAULT_TAGS_FOR_BLOG_POST: Shift the value to the constants.ts file.
(d) WHITELISTED_EXPLORATION_IDS_FOR_PLAYTHROUGHS: Replaced this check to "Is exploration curated".

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->

Config properties are not available on the config-tab anymore:
https://github.com/oppia/oppia/assets/44300735/cbf806c3-f46c-4afb-bd8b-9d7d17a53bf0

Featured translations languages works fine:
![Screenshot from 2023-05-22 10-29-50](https://github.com/oppia/oppia/assets/44300735/0b99a925-eb4d-4841-b694-6baf42130bbc)

When hover over the i button:
![Screenshot from 2023-05-24 12-48-48](https://github.com/oppia/oppia/assets/44300735/5643948c-727d-4efb-848e-5af5e587257c)


List of default tags for blog works fine:
![Screenshot from 2023-05-16 10-34-45](https://github.com/oppia/oppia/assets/44300735/6aaf1717-1398-4457-af3e-224ff346a965)

Blog admin page before the changes: We were allowed to edit the tags.
![Screenshot from 2023-05-24 16-35-55](https://github.com/oppia/oppia/assets/44300735/ac249316-13d7-4b8a-8856-e8af921be02b)

Blog admin page after the changes. The config property to edit the tags have been removed, to edit the tags we have to now edit the constants.ts file.
![Screenshot from 2023-05-24 16-34-45](https://github.com/oppia/oppia/assets/44300735/48b7c652-fee8-4729-a405-784123df199d)


## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

lkbhitesh07 committed:
replaced vmid and classroom config props

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into depricate-few-config-props

lkbhitesh07 committed:
removed featured-translation-languages config property

lkbhitesh07 committed:
removed list-of-default-tags config props

lkbhitesh07 committed:
removed exploration-whitelisted config property

lkbhitesh07 committed:
fixed lint errors

lkbhitesh07 committed:
lint

lkbhitesh07 committed:
fixed test

lkbhitesh07 commented:
Few points:
1. When we create a dummy classroom via the admin page we are not able to see the data on the classroom page, this is because we now show the data on the classroom page from the constants.ts. This is only temporary as we soon are going to shift this property to the classroom-admin page. I'm a bit confused about what should we do for now, should we remove the creation of dummy data for now?
2. We are planning to store `VMID_SHARED_SECRET_KEY_MAPPING` on the cloud secrets and then access it. I have two things to discuss: First, The type of the function [get_secret](https://github.com/oppia/oppia/blob/develop/core/platform/secrets/cloud_secrets_services.py#L39) is of type `Optional[str]` but the value of our secret key will be `List[Dict]` and this results in failing type checks, Just want to ask that we do not store the value that is of type `str` only and we can make changes of the return type of the function.
Second, We have two files [cloud_secrets_services.py](https://github.com/oppia/oppia/blob/develop/core/platform/secrets/cloud_secrets_services.py) and [dev_mode_secrets_services.py](https://github.com/oppia/oppia/blob/develop/core/platform/secrets/dev_mode_secrets_services.py) for prod and dev environments respectively. For the dev mode, we store the secrets in environment variables as mentioned [here](https://github.com/oppia/oppia/blob/develop/core/platform/secrets/dev_mode_secrets_services.py#L38). I'm not sure on how to do that and that is why currently I'm adding a check for the dev mode separately and then assigning the default value. The way we access the secrets in other places is the same for both environments which is something like `secrets_services.get_secret()`. Please let me know how can I add values to environment variables.

Thanks

lkbhitesh07 commented:
@seanlip PTAL

seanlip commented:
Hi @lkbhitesh07,

(1) is a problem -- it will hinder testing and be a problem for new contributors. Can you check with @Nik-09 on a way to avoid this? Maybe we should just migrate directly and skip the intermediate step -- I don't think anything actually blocks that if we don't turn on the diagnostic test feature.

(2) Secrets are strings, so let's break VMID_SHARED_SECRET_KEY_MAPPING into two strings (one per field) and store those as separate secrets.

For the second question, see the PR description [here](https://github.com/oppia/oppia/commit/b4ee6426a57353cf88345e020c6237ad8af1765f), it seems like app.yaml (or app_dev.yaml) is what you want to look at.

seanlip commented on assets/constants.ts:
```diff
@@ -29,6 +29,24 @@ export default {
+  // Add classroom name to SEARCH_DROPDOWN_CLASSROOMS in this file
+  // to add that classroom to learner group syllabus filter whenever a new
+  // classroom is added.
+  "CLASSROOM_PAGES_DATA": [{
```
Having this here might be a problem, see top-level comment. Also I think we need it beyond just for learner groups?

lkbhitesh07 replied:
Restored the classroom config property for now.

seanlip commented on core/controllers/features.py:
```diff
@@ -51,11 +52,10 @@ def get(self, exploration_id: str) -> None:
         self.render_json({
-            'is_exploration_whitelisted':
-                exploration_id in whitelisted_exploration_ids_for_playthroughs,
+            'is_exploration_curated':
```
Prefer exploration_is_curated (for booleans)

lkbhitesh07 replied:
Updated.

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into depricate-few-config-props

lkbhitesh07 committed:
restored classroom config property

lkbhitesh07 committed:
replaced variable name

lkbhitesh07 committed:
lint fix

lkbhitesh07 commented:
@seanlip PTAL

seanlip commented on assets/constants.ts:
```diff
@@ -29,6 +29,13 @@ export default {
   // fragment in the controller is invalid.
   "DEFAULT_CLASSROOM_URL_FRAGMENT": "math",
 
+  "FEATURED_TRANSLATION_LANGUAGES": [],
```
What is this actually used for, and can we get rid of it altogether?

lkbhitesh07 replied:
Hey @seanlip,
Well, we do use this in the contributor dashboard, I have also attached a screenshot in the PR description for reference. Mainly we use this to display some of the most commonly used languages. I think it makes sense to have this functionality but I'm also fine with removing it as the contributor can select the language from the dropdown.
Let me know what do you think.

Thanks

seanlip commented:
@lkbhitesh07 Note that lots of tests are failing, you might want to handle that before getting another review.

For featured translation langs, I have a feeling it's not really being used anywhere. Might be better to remove it altogether if that's the case -- we shouldn't add things to constants.ts if we don't need to. (You can talk to the contributor dashboard team if you want a second opinion on it, though many folks there are busy handling some high-impact breakages at the moment so it might be better to just examine the code directly.)

seanlip commented:
Oh, that's cool, I had no idea we had that, thanks for the screenshot!

Ok let's keep it but add some languages to it -- namely Brazilian
Portuguese, Arabic, Nigerian Pidgin, Spanish, Swahili, Hindi, Hausa, Igbo,
and Yoruba.

On Fri, May 19, 2023, 12:47 PM Hitesh Tomar ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In assets/constants.ts
> <https://github.com/oppia/oppia/pull/18226#discussion_r1198539368>:
>
> > @@ -29,6 +29,13 @@ export default {
>    // fragment in the controller is invalid.
>    "DEFAULT_CLASSROOM_URL_FRAGMENT": "math",
>
> +  "FEATURED_TRANSLATION_LANGUAGES": [],
>
> Hey @seanlip <https://github.com/seanlip>,
> Well, we do use this in the contributor dashboard, I have also attached a
> screenshot in the PR description for reference. Mainly we use this to
> display some of the most commonly used languages. I think it makes sense to
> have this functionality but I'm also fine with removing it as the
> contributor can select the language from the dropdown.
> Let me know what do you think.
>
> Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/18226#discussion_r1198539368>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACQV5SQ7R4BC2FNJDUPPDZDXG33NZANCNFSM6AAAAAAYA4LQNI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

lkbhitesh07 committed:
fixed tests

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into depricate-few-config-props

lkbhitesh07 committed:
fixed lint, type-checking and e2e tests

lkbhitesh07 committed:
fixed backend test

lkbhitesh07 committed:
lint

lkbhitesh07 committed:
made changes to fix the tests

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into depricate-few-config-props

lkbhitesh07 committed:
fixed lints and backend test

lkbhitesh07 committed:
lint fix - double quote to single

lkbhitesh07 committed:
restored the spaces

lkbhitesh07 commented:
@seanlip PTAL

seanlip commented on assets/constants.ts:
```diff
@@ -29,6 +29,22 @@ export default {
   "DEFAULT_CLASSROOM_URL_FRAGMENT": "math",
 
+  "FEATURED_TRANSLATION_LANGUAGES": [
+    {"language_code": "pt", "explanation": "português (Portuguese)"},
```
Wouldn't just the language codes be enough (and we call this FEATURED_TRANSLATION_LANGUAGE_CODES)? The descriptions should be retrieved from the existing languages dict, and for safety we should have a test that verifies that every language code in this list exists in the existing languages dict.

That way you don't have multiple sources of truth for the explanation.

lkbhitesh07 replied:
We can do that but please note that currently also we do not use the "explanation" as the "description". Just to explain it more, we fetch the description of the language code from the [constants.SUPPORTED_AUDIO_LANGUAGES](https://github.com/oppia/oppia/blob/develop/assets/constants.ts#L5355) and the "explanation" is used when the user hovers over the "i" button present over the language.
Currently, I have put in the same value as the "description" to the "explanation" because I was not sure what to put there.
Let me know if we plan to remove the "explanation" part.

I have added the test as requested in the "contributor_dashboard_test.py" file.

Thanks

seanlip replied:
Ah, I remember what this was for now! You can keep the 'explanation' then and use the following as explanations instead (and, next time, please ask if you're not sure what to put, rather than assume; we could avoid quite a lot of back-and-forth that way):

- pt: "For learners in Brazil, Angola and Mozambique."
- ar: "For learners in Arabic-speaking countries in the Middle East."
- pcm: "For learners in Nigeria."
- es: "For learners in Latin America and South America."
- sw: "For learners in Kenya and Tanzania."
- hi: "For learners in India."
- ha: "For learners in Nigeria."
- ig: "For learners in Nigeria."
- yo: "For learners in Nigeria."

lkbhitesh07 replied:
Got it, thanks for the info.
I will make sure to ask if I'm not sure.
Thanks

seanlip commented on core/templates/components/state-directives/outcome-editor/outcome-destination-editor.component.spec.ts:
```diff
@@ -79,7 +79,7 @@ describe('Outcome Destination Editor', () => {
     );
 
-    spyOn(stateEditorService, 'isExplorationWhitelisted').and.returnValue(true);
+    spyOn(stateEditorService, 'getExplorationIsCurated').and.returnValue(true);
```
isExplorationCurated

lkbhitesh07 replied:
Updated

seanlip reviewed (changes requested):
Thanks! A couple of comments.

seanlip commented:
Also adding @JayVivarekar @kevintab95 @gp201 @chris7716 for codeowners.

lkbhitesh07 committed:
added test to check the list of featured translations

lkbhitesh07 committed:
corrected the naming convention

lkbhitesh07 committed:
fixed the typos

lkbhitesh07 commented:
@seanlip PTAL

lkbhitesh07 committed:
updated featured-translation-languages part

lkbhitesh07 committed:
fixed lints

lkbhitesh07 committed:
length lint fix

lkbhitesh07 commented on core/domain/classifier_services.py:
```diff
@@ -94,10 +97,15 @@ def verify_signature(
-        if val['vm_id'] == oppia_ml_auth_info.vm_id:
-            secret = val['shared_secret_key'].encode('utf-8')
-            break
+    vm_id = secrets_services.get_secret('VM_ID')
+    shared_secret_key = secrets_services.get_secret('SHARED_SECRET_KEY')
+    if (vm_id is None or shared_secret_key is None) and constants.DEV_MODE:
+        vm_id = 'vm_default'
+        shared_secret_key = '1a2b3c4e'
+    assert shared_secret_key is not None
+    assert vm_id is not None
+    if vm_id == oppia_ml_auth_info.vm_id:
+        secret = shared_secret_key.encode('utf-8')
```
Hey @seanlip,
So as per our previous discussion, we were planning to add the secret to the `app_dev.yaml` file so that we can get them via `secrets_services.get_secret()` function in the dev mode but after looking at other tests I found out that we are mocking the `get_secret` function in order to test out the things. Please note that when I tried to get the vmid secrets via `secrets_services.get_secret` I was getting None, for this I edited the `app_dev.yaml` file(you can find the changes at the top of the file structure). When I tried to print the value of `os.environ` I was not able to locate the `SECRETS` environment variable.

I have added a check for the dev mode in the changes above in order to pass the default value.

seanlip replied:
Sorry, I'm confused: this is not a test file but you're talking about tests here.

Also, what is your question exactly?

lkbhitesh07 replied:
Oh, I just wanted to ask if you are fine with the changes above.
Yes, this is not the test file, just wanted to bring your attention that the `get_secret` function is currently not working as expected for the dev environment as instead of getting the secrets from the environment variable we have to mock the function in order to pass the tests. Here in the file, I'm adding a specific check for the dev environment in order to pass the default value but instead, we would be getting the value from the environment variable.
Thanks

seanlip replied:
Ah I see. No, don't do this -- let's not introduce dev-mode conditionals into the main code just to help the tests. Do explicit mocking for the tests if needed.

lkbhitesh07 replied:
But this condition is not only for the tests, if I remove this then we will get an error in the local development server when we try to get the vmid secrets. What should we do in that case?

EDIT: I think we will not get the error instead we will get the None value. Can you tell me what we plan to do in the development mode? I will apply the `assert` accordingly. The reason I think the current changes make sense is that this handles both the dev part and the prod part.

seanlip replied:
What do we currently do for other secrets?

Generally we should have a process for handling secrets in dev mode that actually works. Is this code actually exercisable in dev mode when using the server normally? If so, I agree with you that we need to find a way, but if not, maybe we can file an issue for that and leave it to later.

If we do need to fix it, perhaps search and read up about how cloud secrets are supposed to be handled in dev mode -- we can't be the first folks to run into this issue.

lkbhitesh07 replied:
I took a look at other secrets, we are simply using them via `secrets_services.get_secret()` without any conditions.
We can close this discussion as I was able to access the secrets when the dev server was on. It now makes sense why we use the swapping for the `get_secret` function in the tests as we cannot really access the environment variables when running the tests.

I have updated the tests and the functionality works fine.

lkbhitesh07 commented:
Hey @seanlip, I have made the changes. Please have another pass.
Please note that I have added a [comment](https://github.com/oppia/oppia/pull/18226/files#r1201499264) and require your suggestion.
@seanlip PTAL

seanlip commented on core/tests/webdriverio_desktop/contributorDashboard.js:
```diff
@@ -515,15 +505,23 @@ describe('Translation contribution featured languages', () => {
 
   it('should show correct featured languages', async function() {
+    let featuredLanguages = [];
+    for (let language in Constants.FEATURED_TRANSLATION_LANGUAGES) {
```
This test has too much logic in it. You should hardcode the expected list.

lkbhitesh07 replied:
Done

seanlip commented on core/tests/webdriverio_desktop/contributorDashboard.js:
```diff
@@ -515,15 +505,23 @@ describe('Translation contribution featured languages', () => {
       .mouseoverFeaturedLanguageTooltip(0);
     await contributorDashboardTranslateTextTab
-      .expectFeaturedLanguageExplanationToBe('Partnership with ABC');
+      .expectFeaturedLanguageExplanationToBe('português (Portuguese)');
```
This seems wrong?

lkbhitesh07 replied:
Yes, Updated.

seanlip commented on core/domain/config_domain.py:
```diff
@@ -147,24 +146,6 @@ class ConfigPropertySchemaDict(TypedDict):
     }
 }
 
-VMID_SHARED_SECRET_KEY_SCHEMA = {
```
Please update the comment in line 215 with the deleted config property names.

lkbhitesh07 replied:
Done

seanlip commented on core/controllers/contributor_dashboard_test.py:
```diff
@@ -1398,25 +1398,65 @@ class FeaturedTranslationLanguagesHandlerTest(test_utils.GenericTestBase):
 
     def test_get_featured_translation_languages(self) -> None:
         response = self.get_json('/retrivefeaturedtranslationlanguages')
```
retrive --> retrieve

lkbhitesh07 replied:
Done

lkbhitesh07 committed:
done suggested changes

lkbhitesh07 committed:
lint fix

lkbhitesh07 committed:
fixed mypy

lkbhitesh07 commented:
Done with the changes @seanlip.
Thanks

seanlip commented on core/controllers/acl_decorators_test.py:
```diff
@@ -7113,8 +7113,29 @@ def extract_request_message_vm_id_and_signature(
         def post(self) -> None:
             self.render_json({'job_id': 'new_job'})
 
+    def _swap_function(self, name: str) -> Optional[str]:
```
This should be called something like _mock_get_secret(). The name you're using here is too generic.

lkbhitesh07 replied:
Updated

seanlip commented on core/controllers/acl_decorators_test.py:
```diff
@@ -7113,8 +7113,29 @@ def extract_request_message_vm_id_and_signature(
+
     def setUp(self) -> None:
         super().setUp()
+        self.swap_secret = self.swap_with_checks(
```
Don't do this here. Only do it in specific tests which need the swap.

lkbhitesh07 replied:
Updated

seanlip commented on core/domain/classifier_services.py:
```diff
@@ -94,10 +96,12 @@ def verify_signature(
-            break
+    vm_id = secrets_services.get_secret('VM_ID')
+    shared_secret_key = secrets_services.get_secret('SHARED_SECRET_KEY')
+    assert shared_secret_key is not None
```
Just to check, this function doesn't get called on general usage, right? (So we can avoid setting up these secrets in prod until we actually need the functionality?)

lkbhitesh07 replied:
Yeah, kind of. This `verify_signature` function gets used in one of the acl_decorators which is `is_from_oppia_ml` and this decorator gets used in 2 handler's post requests but currently we do not have any post request in our codebase to these handlers.
Please note that when we plan to use this functionality we must have secrets stored in the cloud as I have added `assert` checks for secrets having not None value.

seanlip commented on core/domain/classifier_services_test.py:
```diff
@@ -719,6 +720,13 @@ def test_generate_signature(self) -> None:
     def test_verify_signature(self) -> None:
         """Test the verify_signature method."""
 
+        def _swap_function(name: str) -> Optional[str]:
```
Same note about naming.

lkbhitesh07 replied:
Updated

seanlip commented on core/tests/webdriverio_desktop/contributorDashboard.js:
```diff
@@ -515,15 +505,27 @@ describe('Translation contribution featured languages', () => {
   });
 
   it('should show correct featured languages', async function() {
+    let featuredLanguages = [
```
expectedFeaturedLanguages is probably better (you can use the "expected" prefix to distinguish expected from actual).

lkbhitesh07 replied:
Updated

seanlip reviewed (changes requested):
Thanks @lkbhitesh07, took a full pass. Please fix the backend tests too.

Also, please also provide a screenshot of the latest version of the featured translations dropdown, where the "i" icon is hovered over, since this is the first time we're actually using it. Thanks!

seanlip reviewed (changes requested):
Thanks @lkbhitesh07, took a full pass. Please fix the backend tests and frontend tests too.

Also, please also provide a screenshot of the latest version of the featured translations dropdown, where the "i" icon is hovered over, since this is the first time we're actually using it. Thanks!

lkbhitesh07 committed:
fixed tests and did suggested changes

lkbhitesh07 committed:
fixed lint

lkbhitesh07 committed:
fixed backend and e2e test

lkbhitesh07 committed:
fixed lint

lkbhitesh07 committed:
nit changes in backend test

lkbhitesh07 commented:
> Thanks @lkbhitesh07, took a full pass. Please fix the backend tests and frontend tests too.
> 
> Also, please also provide a screenshot of the latest version of the featured translations dropdown, where the "i" icon is hovered over, since this is the first time we're actually using it. Thanks!

Hey @seanlip I have added the screenshot in the description above. I have also fixed all the tests in case any backend test reports I will update it asap.
Please have another pass.

Thanks

lkbhitesh07 commented:
@seanlip PTAL

lkbhitesh07 commented:
Hey @JayVivarekar, @kevintab95 @gp201 @chris7716 Just a reminder for the review.
Thanks

seanlip commented on core/controllers/classifier_test.py:
```diff
@@ -143,11 +145,32 @@ def setUp(self) -> None:
+
     def test_trained_classifier_handler(self) -> None:
         # Normal end-to-end test.
-        self.post_blob(
```
Can you explain the need for the changes in this file -- why do we need the mock (or the secret values) now when we didn't before?

lkbhitesh07 replied:
Sure, So the `get_secret` function gets used in a decorator from the `acl_decorators` file which is `is_from_oppia_ml`. Now this decorator gets used in 2 handlers which are `TrainedClassifierHandler` and `NextJobHandler`. The mock changes that are recently added are part of these handlers.
I think I might've missed adding the tests before as I was adding several other tests at other places.

seanlip replied:
Yup, but why didn't these tests fail before?

lkbhitesh07 replied:
They were failing before as well. As we have a bunch of backend tests and if any of the test fail from the queue then the rest of the backend tests gets canceled, that is why I think these were not reported before, and as I was fixing the tests the rest got reported.
If this is what you are asking here :).

Thanks

seanlip replied:
No, I meant, why aren't they failing in develop?

lkbhitesh07 replied:
I and Sean discussed this in the data-stability team meeting. Just to update the thread, Previously the reason why we did not require to mock the secrets was that we were using the vmid-secrets config properties which were returning the default value at that time and now we have removed the config property and decided to go ahead with the cloud secrets and that is why we require to mock the tests here as part of the development server.

Thanks

seanlip commented on core/tests/puppeteer-acceptance-tests/spec/blog-admin-tests/assign-roles-to-users-and-change-tag-properties.spec.js:
```diff
@@ -60,10 +60,6 @@ describe('Blog Admin', function() {
       await superAdmin.expectUserNotToHaveRole(
         'guestUsr2', ROLE_BLOG_POST_EDITOR);
 
-      await blogAdmin.expectTagToNotExistInBlogTags('Test_Tag');
```
Does the functionality for the blog admin to do this still exist? What does their admin page look like now?

lkbhitesh07 replied:
I'm not sure if this existed before the changes as well. Have a look at the before and after changes:

before:
![Screenshot from 2023-05-24 16-35-55](https://github.com/oppia/oppia/assets/44300735/c8115424-29a6-4b9b-ae9e-6044f4266105)

after:
![Screenshot from 2023-05-24 16-34-45](https://github.com/oppia/oppia/assets/44300735/1fe28c61-2bdd-4775-97de-2a808d606720)

As we are no longer using the blog tags config property I have removed it.
I have also updated the description with the screenshots

Thanks

seanlip replied:
This all seems good. Thanks!

gp201 replied:
Just to be sure, Are we removing the tag functionality, or are we giving users the flexibility to create any tag?

lkbhitesh07 replied:
Hey @gp201, We are not removing the tag functionality instead now we have a fixed set of tags that will be available for the users to select from. We have a constant defined in `constants.ts` by the name `LIST_OF_DEFAULT_TAGS_FOR_BLOG_POST` and if in the future we plan to introduce some new tags we will simply be updating the value of LIST_OF_DEFAULT_TAGS_FOR_BLOG_POST.

Thanks

seanlip reviewed (changes requested):
Thanks @lkbhitesh07, took another pass!

lkbhitesh07 committed:
fixed the backend test

seanlip commented:
@lkbhitesh07 Replied to the first comment, PTAL: https://github.com/oppia/oppia/pull/18226#discussion_r1203877763

Also backend tests are still failing, FYI.

lkbhitesh07 commented:
> @lkbhitesh07 Replied to the first comment, PTAL: [#18226 (comment)](https://github.com/oppia/oppia/pull/18226#discussion_r1203877763)
> 
> Also backend tests are still failing, FYI.

Thanks @seanlip I have [replied](https://github.com/oppia/oppia/pull/18226#discussion_r1203978171).
I have pushed the fix for the backend test.
Thanks

JayVivarekar reviewed (approved):
LGTM for codeowner files

lkbhitesh07 commented:
Hey @seanlip I have replied to the comment [here](https://github.com/oppia/oppia/pull/18226#discussion_r1203877763). Also, just a ping in case you missed this comment [here](https://github.com/oppia/oppia/pull/18226#discussion_r1202245426).

@seanlip PTAL

lkbhitesh07 commented:
Hey @kevintab95 @gp201 @chris7716, sorry for the rush, but we actually plan to wrap this PR up by tomorrow. I was wondering if you guys can provide review by today, I can make the changes if needed by today only so that we can merge this by tomorrow.

Thanks

seanlip reviewed (approved):
LGTM for codeowners. Thanks!

chris7716 reviewed (approved):
LGTM for codeowner files.

gp201 commented on core/controllers/contributor_dashboard_test.py:
```diff
@@ -1397,26 +1397,66 @@ class FeaturedTranslationLanguagesHandlerTest(test_utils.GenericTestBase):
+                    'explanation': 'For learners in Nigeria.'
+                }]
+        }
+        self.assertEqual(response, expected_response)
```
Can we use `constants.FEATURED_TRANSLATION_LANGUAGES` instead?

lkbhitesh07 replied:
Well, we can but previously Sean mentioned that in tests we should hardcode the value to check the output if it matches or not.
Let me know what you think.

Thanks

gp201 reviewed (changes requested):
Thanks, @lkbhitesh07 ! I left some comments. PTAL

lkbhitesh07 commented:
> Thanks, @lkbhitesh07 ! I left some comments. PTAL

Thanks for the review @gp201, I have responded.
@gp201 PTAL

lkbhitesh07 commented:
Hey @gp201, Just a friendly ping for a review in case you missed my previous comment.

Thanks

gp201 reviewed (approved):
LGTM
